### ISSUE TITLE
Some visual feedback is needed (progress bar?) while uploading document to be used for RAG

### ISSUE THREAD

**Issue by kha84** [open-webui/open-webui#2190]
**Is your feature request related to a problem? Please describe.**
I'm thrilled to see the RAG feature has finally came to my beloved web UI. But when I upload some big documents (say 15 Mb of PDF) I get no visual feedback on what is happening - the front-end doesn't give any clues and the file I have just submitted doesn't show anywhere until it's finally processed. I know such big documents take a while to be processed - and I see that from docker logs (open-web-ui is installed via docker), but this is the only how I can track the progress of creating embeddings. And only once the document is processed it appears in the list - which is the only visual indication I get from UI.

**Describe the solution you'd like**
It would be great to see some additional visual indicator showing a processing progress, denoting that nothing has stucked or frozen, but it's still being processed, after new document was added. Ideally with some kind of progress bar. I found a place in frontend code where it's done, but my JS/TS knowledge is limited to make proper changes - https://github.com/open-webui/open-webui/blob/90503be2edef1a1f7ce2074286b6316d5cb8868a/src/lib/apis/rag/index.ts#L167

**Describe alternatives you've considered**
If there's no easy way to get a progress indicator, at least to introduce some modal window saying "please wait till your document will be processed"

**Comment by AdaptiveStep** [comment]
Yeah, maybe a little "miniwizard" for importing documents would be in order. It seems to completely ignore the content of the PDF if it only contains one image per page. This way it could show how much storage it takes, and so on..

**Comment by tjbck** [comment]
Closing in favour of #3520
